src/SubscriberBase.h

```diff
@@ -83,6 +83,15 @@ class SubscriberBaseT : public Subscriber<T>,
       bool startExecutor = true)
       : ExecutorBase(executor, startExecutor), cancelled_(false) {}
 
+  virtual ~SubscriberBaseT() {
+    if (!cancelled_ && originalSubscription_) {
```

Comment:
When you say the SubscriberBaseT is about to be destroyed (because nobody is using it anymore) that should mean somebody sent a terminating signal to that instance prior releasing the shared_ptr. And if I am looking correctly at the code, if you call any of the {cancel, onComlete, onError} then the subscription is always going to be cancelled.
If the Subscription is not cancelled then the code never sent the terminating signal. And based on the contract, this is a bug.
I agree that if you lost the SubscriberBaseT instance already, nobody will be able to call cancel on the subscription because the executor is also gone.